### PR TITLE
[ricos-content] improve(convertors): spoiler, mention, anchor

### PR DESCRIPTION
**Pull request by Barackos**
1. HTML Convertor covers `anchor` links
2. HTML Convertor supports spoiler & mention to cover all textual elements for the Translatables API.

### CODE REVIEW

File: packages/ricos-content/web/src/converters/html/fromHtml/core/rules.ts [hunk 1]
```diff
@@ -116,6 +119,26 @@ export const colorStyleToTextColor: Rule = [
     ),
 ];
 
+// NOTE: Created for translations functionality.
```
Comment (lxgreen) [inline packages/ricos-content/web/src/converters/html/fromHtml/core/rules.ts:122]: why not applicable? it returns valid mention content

File: packages/ricos-content/web/src/converters/html/fromHtml/core/rules.ts [hunk 2]
```diff
@@ -116,6 +119,26 @@ export const colorStyleToTextColor: Rule = [
...
+  ]),
+  context => (node: Element) => {
+    const { 'data-mention-name': name, 'data-mention-slug': slug } = getAttributes(node);
+    return context.addDecoration(Decoration_Type.MENTION, { mentionData: { name, slug } }, node);
+  },
+];
+
+// NOTE: Created for translations functionality.
+// Not applicable for display.
+export const spanToSpoiler: Rule = [
```
Comment (lxgreen) [inline packages/ricos-content/web/src/converters/html/fromHtml/core/rules.ts:137]: see comment regarding `spoiler=true`

File: packages/ricos-content/web/src/converters/html/toHtml/toHtml.ts [hunk 3]
```diff
@@ -32,24 +41,60 @@ const nodeToHtml = (node: Node): string | string[] => {
...
-      return style ? { style } : {};
-    }
-    default:
-      return {};
+const createLinkAttrs = (decoration: Decoration): Record<string, string> => {
+  const rel = Object.entries(decoration.linkData?.link?.rel || {})
+    .filter(([_, value]) => !!value)
+    .map(([key]) => key)
+    .join(' ');
+  const m_target = decoration.linkData?.link?.target;
```
Comment (lxgreen) [inline packages/ricos-content/web/src/converters/html/toHtml/toHtml.ts:49]: C++? what's m_name?

File: packages/ricos-content/web/src/converters/html/toHtml/toHtml.ts [hunk 4]
```diff
@@ -32,24 +41,60 @@ const nodeToHtml = (node: Node): string | string[] => {
...
+  return pickBy({ href, rel, anchor, target }, identity);
+};
+
+const createAnchorAttrs = (decoration: Decoration): Record<string, string> => {
+  const anchor = decoration.anchorData?.anchor || '';
+  return { href: `#${anchor}`, rel: 'noopener noreferrer' };
+};
+
+// NOTE: Created for translations functionality.
+// Not applicable for display.
```
Comment (lxgreen) [inline packages/ricos-content/web/src/converters/html/toHtml/toHtml.ts:64]: irrelevant comment, maybe better to note that the mention html representation is not interactive

File: packages/ricos-content/web/src/converters/html/toHtml/toHtml.ts [hunk 5]
```diff
@@ -32,24 +41,60 @@ const nodeToHtml = (node: Node): string | string[] => {
...
+const createMentionAttrs = (decoration: Decoration): Record<string, string> => {
+  const { name = '', slug = '' } = decoration.mentionData || {};
+  if (!name && !slug) {
+    return {};
   }
+  return pickBy({ role: 'link', 'data-mention-name': name, 'data-mention-slug': slug }, identity);
 };
 
+// NOTE: Created for translations functionality.
+// Not applicable for display.
```
Comment (lxgreen) [inline packages/ricos-content/web/src/converters/html/toHtml/toHtml.ts:74]: also here

**Review by lxgreen — CHANGES_REQUESTED** [review]

**Comment by Barackos** [comment]
Refactoring here: #2901

File: packages/ricos-content/web/__tests__/translatablesMock.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
...
+    "text": "<p><strong>Lorem ipsum dolor sit amet</strong> 😎, <u>consectetur adipiscing</u> elit. Quisque quis ex <em>auctor</em>, facilisis lectus id, sodales ipsum. In <strong>sagittis </strong><u><strong>ultrices</strong></u><u> sem</u> ac elementum. Ut <u><a href=\"www.google.com\" rel=\"nofollow sponsored ugc\" target=\"_blank\">ut</a></u> iaculis ex. <span spoiler=\"true\">Lorem ipsum</span> dolor sit <span role=\"link\" data-mention-name=\"amet\" data-mention-slug=\"amet\">@amet</span>, consectetur adipiscing elit. In hac <span style=\"color: color4;\">habitasse </span><span style=\"color: color4;\"><span style=\"background-color: color3;\">platea</span></span><span style=\"background-color: color3;\"> dictumst</span>. Aenean placerat, quam eget vestibulum <u><a href=\"www.link.com\" target=\"_blank\">varius</a></u>, nisl ligula vulputate <u><a href=\"#98ccj\" rel=\"noopener noreferrer\">tortor</a></u>, id lacinia odio magna eu arcu.</p>"
```
[the remark this answers is not in the document]
Comment (Barackos) [reply packages/ricos-content/web/__tests__/translatablesMock.json:5]: I chose `data-spoiler`. #2901

File: packages/ricos-content/web/src/converters/html/fromHtml/core/rules.ts [hunk 7]
```diff
@@ -56,9 +56,12 @@ export const aToLink: Rule = [
   hasTag('a'),
   context => (node: Element) => {
     const attrs = getAttributes(node);
+    const isAnchor = attrs.href.startsWith('#');
```
[the remark this answers is not in the document]
Comment (Barackos) [reply packages/ricos-content/web/src/converters/html/fromHtml/core/rules.ts:59]: Modified it so that ruling will force a truthy `href` value. #2901

**Pull request merged**